### PR TITLE
z64player.h and Player struct work

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

- Moved the existing `Player` struct to a new file, `z64player.h`, and name a lot of the variables (thanks to @Zelllll , @engineer124 and @Thar0 for most of the names and new documentation). 
- Replaced symbols everywhere.
- Named `func_8012403C` to `Player_GetMask` and use mask macros. Also named the unused `func_8012404C` below it to `Player_RemoveMask`.
- Macros for `transformation` and `currentShield`
- Some minor cleanup work on some actors, added macros for collision flags where they were missing.

I have not gone into `itemId` or `actionParam` documentation in the interests of limiting the scope of this PR. Apologies in advance to everyone who this causes conflicts for, including me: I tried to touch as few files as possible.

### CODE REVIEW

**Review by engineer124 — COMMENTED** [review]
Nice. Just remove a single comment and this is good to go from me

**Review by engineer124 — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
thanks for doing this!

**Review by isghj5 — APPROVED** [review]
Found nothing wrong, matches all my unfinished actors.

I did see one spot where a non-matching function is using an old player variable name that was changed, but that will get fixed with the function.

Thanks for doing this

**Comment by EllipticEllipsis** [comment]
That should be corrected too, though, in case someone wants to build non-matching. Which function is it?

File: src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c [hunk 1]
```diff
@@ -240,18 +245,18 @@ void func_80931004(EnTuboTrap* this, GlobalContext* globalCtx) {
...
         startingRotation = this->actor.home.rot.z;
         if ((startingRotation == 0) || (this->actor.yDistToPlayer <= ((f32)startingRotation * 10.0f))) {
             func_800BC154(globalCtx, &globalCtx->actorCtx, this, ACTORCAT_ENEMY);
             currentHeight = this->actor.world.pos.y;
             this->actor.flags |= 0x11; // always update and can target
 
-            // hard to know what this value is even used for without knowing what ActorPlayer::unk14B is
+            // hard to know what this value is even used for without knowing what Player::unk14B is
             // wild guess: this is player animation state, height is modified to always point at center of link model
             weirdvalue = D_8093146C[player->unk14B & 0xFF];
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c:257]: This value, this becomes player->transformation

and the speculative comment above which was correct I guess

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c:257]: Yeah, I'll just remove the comment and change it.

**Review by Kenix3 — APPROVED** [review]
Looks good to me. Will merge once the conflicts are resolved.

**Pull request merged**